sentinelsat/sentinel.py

```diff
@@ -318,28 +315,42 @@ def download(self, id, directory_path='.', checksum=False, check_existing=False)
 
         self.logger.info('Downloading %s to %s' % (id, path))
 
-        # Check if the file exists and passes md5 test
-        # The download function will by default continue the download if the file exists but is incomplete
-        if exists(path) and getsize(path) == product_info['size']:
-            if not check_existing or _md5_compare(path, product_info['md5']):
-                self.logger.info('%s was already downloaded.' % path)
-                return product_info
+        if exists(path):
```

Comment:
Why abandon the feature of `check_existing`? I find it very useful to test the integrity of files I am given or acquired through other means than the Open Access Hub. check_existing=True is a feature that allows me to quickly compare a folder of files to the MD5 checksums published at the DHuS.

It could be easily implemented at this point, with the current logic beeing moved to an `elif`.